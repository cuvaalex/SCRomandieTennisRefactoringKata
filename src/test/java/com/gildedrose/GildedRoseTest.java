package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.Test;

public class GildedRoseTest {

    @Test
    public void combinationUpdateQuality() {
        String[] names = new String[]{"Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "blablabla"};
        Integer[] sellIns = new Integer[]{-5, -1, 0, 1, 5, 6, 10, 11};
        Integer[] qualities = new Integer[]{0, 1, 20, 49, 50};
        CombinationApprovals.verifyAllCombinations(
                (name, sellIn, quality) -> {
                    Item item = new Item(name, sellIn, quality);
                    GildedRose gildedRose = new GildedRose(new Item[]{item});

                    gildedRose.updateQuality();
                    return item.toString();
                }, names, sellIns, qualities);
    }
}