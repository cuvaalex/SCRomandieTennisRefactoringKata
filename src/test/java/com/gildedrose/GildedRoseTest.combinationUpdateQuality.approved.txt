[Aged Brie, -5, 0] => Aged Brie, -6, 2 
[Aged Brie, -5, 1] => Aged Brie, -6, 3 
[Aged Brie, -5, 20] => Aged Brie, -6, 22 
[Aged Brie, -5, 49] => Aged Brie, -6, 50 
[Aged Brie, -5, 50] => Aged Brie, -6, 50 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 20] => Aged Brie, -2, 22 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 20] => Aged Brie, -1, 22 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 1] => Aged Brie, 0, 2 
[Aged Brie, 1, 20] => Aged Brie, 0, 21 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 20] => Aged Brie, 4, 21 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 1] => Aged Brie, 5, 2 
[Aged Brie, 6, 20] => Aged Brie, 5, 21 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 20] => Aged Brie, 9, 21 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 20] => Aged Brie, 10, 21 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, -5, 0] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 1] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 20] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 49] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 50] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 20] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 20] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Backstage passes to a TAFKAL80ETC concert, 0, 4 
[Backstage passes to a TAFKAL80ETC concert, 1, 20] => Backstage passes to a TAFKAL80ETC concert, 0, 23 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 20] => Backstage passes to a TAFKAL80ETC concert, 4, 23 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Backstage passes to a TAFKAL80ETC concert, 6, 20] => Backstage passes to a TAFKAL80ETC concert, 5, 22 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 20] => Backstage passes to a TAFKAL80ETC concert, 9, 22 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 20] => Backstage passes to a TAFKAL80ETC concert, 10, 21 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[blablabla, -5, 0] => blablabla, -6, 0 
[blablabla, -5, 1] => blablabla, -6, 0 
[blablabla, -5, 20] => blablabla, -6, 18 
[blablabla, -5, 49] => blablabla, -6, 47 
[blablabla, -5, 50] => blablabla, -6, 48 
[blablabla, -1, 0] => blablabla, -2, 0 
[blablabla, -1, 1] => blablabla, -2, 0 
[blablabla, -1, 20] => blablabla, -2, 18 
[blablabla, -1, 49] => blablabla, -2, 47 
[blablabla, -1, 50] => blablabla, -2, 48 
[blablabla, 0, 0] => blablabla, -1, 0 
[blablabla, 0, 1] => blablabla, -1, 0 
[blablabla, 0, 20] => blablabla, -1, 18 
[blablabla, 0, 49] => blablabla, -1, 47 
[blablabla, 0, 50] => blablabla, -1, 48 
[blablabla, 1, 0] => blablabla, 0, 0 
[blablabla, 1, 1] => blablabla, 0, 0 
[blablabla, 1, 20] => blablabla, 0, 19 
[blablabla, 1, 49] => blablabla, 0, 48 
[blablabla, 1, 50] => blablabla, 0, 49 
[blablabla, 5, 0] => blablabla, 4, 0 
[blablabla, 5, 1] => blablabla, 4, 0 
[blablabla, 5, 20] => blablabla, 4, 19 
[blablabla, 5, 49] => blablabla, 4, 48 
[blablabla, 5, 50] => blablabla, 4, 49 
[blablabla, 6, 0] => blablabla, 5, 0 
[blablabla, 6, 1] => blablabla, 5, 0 
[blablabla, 6, 20] => blablabla, 5, 19 
[blablabla, 6, 49] => blablabla, 5, 48 
[blablabla, 6, 50] => blablabla, 5, 49 
[blablabla, 10, 0] => blablabla, 9, 0 
[blablabla, 10, 1] => blablabla, 9, 0 
[blablabla, 10, 20] => blablabla, 9, 19 
[blablabla, 10, 49] => blablabla, 9, 48 
[blablabla, 10, 50] => blablabla, 9, 49 
[blablabla, 11, 0] => blablabla, 10, 0 
[blablabla, 11, 1] => blablabla, 10, 0 
[blablabla, 11, 20] => blablabla, 10, 19 
[blablabla, 11, 49] => blablabla, 10, 48 
[blablabla, 11, 50] => blablabla, 10, 49 
